### PR TITLE
WIP: Refactor Context and Tags

### PR DESCRIPTION
**Pull request by zhangyelong**
Resolves: #2004

### CODE REVIEW

File: app/protocol/constants.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 
 const (
 	// all store name
+	BankStroe            = "bank"
```
Comment (chengwenxi) [inline app/protocol/constants.go]: Nothing is stored in the bank module, so it's seems not need to define it.

File: app/v1/asset/internal/keeper/ante_test.go [hunk 2]
```diff
@@ -57,42 +57,51 @@ func TestAssetAnteHandler(t *testing.T) {
...
-	acc1.SetCoins(sdk.Coins{gatewayCreateFee.Add(nativeTokenIssueFee)})
+	_ = acc1.SetCoins(sdk.Coins{gatewayCreateFee.Add(nativeTokenIssueFee)})
+	ak.SetAccount(newCtx, acc1)
 	_, res, abort := anteHandler(newCtx, tx1, false)
 	require.Equal(t, true, abort)
 	require.Equal(t, false, res.IsOK())
 
-	// assert that the ante handler will return with `abort` set to true
-	acc1.SetCoins(acc1.GetCoins().Add(sdk.Coins{gatewayTokenIssueFee}))
+	// assert that the ante handle`r will return with `abort` set to true
```
Comment (chengwenxi) [inline app/v1/asset/internal/keeper/ante_test.go]: handler

File: app/v1/stake/keeper/delegation.go [hunk 3]
```diff
@@ -528,7 +528,8 @@ func (k Keeper) CompleteUnbonding(ctx sdk.Context, delAddr sdk.AccAddress, valAd
 		return types.ErrNoUnbondingDelegation(k.Codespace())
 	}
 	if !ubd.Balance.IsZero() {
-		ctx.CoinFlowTags().AppendCoinFlowTag(ctx, valAddr.String(), ubd.DelegatorAddr.String(), ubd.Balance.String(), sdk.UndelegationFlow, ubd.TxHash)
+		//ctx.TagsManager().AddCoinFlow(ctx, valAddr.String(), ubd.DelegatorAddr.String(), ubd.Balance.String(), sdk.UndelegationFlow, ubd.TxHash)
```
Comment (chengwenxi) [inline app/v1/stake/keeper/delegation.go]: remove

**Comment by chengwenxi** [comment]
fix https://github.com/irisnet/irishub/issues/2019

File: app/v1/asset/internal/keeper/ante.go [hunk 4]
```diff
@@ -78,3 +72,14 @@ func NewAnteHandler(k Keeper) sdk.AnteHandler {
 		return newCtx, sdk.Result{}, false
 	}
 }
+
+func getSignerAccs(ctx sdk.Context, am auth.AccountKeeper, addrs []sdk.AccAddress) (accs []auth.Account, res sdk.Result) {
```
Comment (zhiqiang-bianjie) [inline app/v1/asset/internal/keeper/ante.go]: more gas consumed may cause consensus failure

File: mock/baseapp/baseapp.go [hunk 5]
```diff
@@ -436,7 +436,11 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 	} else {
 		// In the first block, app.deliverState.ctx will already be initialized
 		// by InitChain. Context is now updated with Header information.
-		app.deliverState.ctx = app.deliverState.ctx.WithBlockHeader(req.Header).WithBlockHeight(req.Header.Height)
+		app.deliverState.ctx = app.deliverState.ctx.
+			WithBlockHeader(req.Header).
+			WithBlockHeight(req.Header.Height).WithCheckValidNum(sdk.NewValidTxCounter()).
```
Comment (zhiqiang-bianjie) [inline mock/baseapp/baseapp.go]: Field `Height` has existed in Header, and `WithBlockHeight` should be removed

**Pull request closed without merging**